src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -99,6 +101,27 @@ public void Prepare(CancellationToken token)
                     info.NewRuleSetFilePath = newRuleSetFilePath;
                 }
             }
+
+            EnsureNoConflictingAdditionalFile();
+        }
+
+        private void EnsureNoConflictingAdditionalFile()
+        {
+            var projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
+            projectSystem.AssertLocalServiceIsNotNull();
+
+            // If the correct file is already in the project, it means we were successful in adding it and there is no clash
+            if (!projectSystem.IsFileInProject(initializedProject, cSharpVBBindingConfig.AdditionalFile.Path))
+            {
+                var additionalFileName = Path.GetFileName(cSharpVBBindingConfig.AdditionalFile.Path);
+
+                var hasConflict = additionalFileConflictChecker.HasConflictingAdditionalFile(initializedProject, additionalFileName, out var conflictingAdditionalFilePath);
+
+                if (hasConflict)
+                {
+                    throw new Exception($"Could not bind project {initializedProject.FullName}: {Path.GetFileName(cSharpVBBindingConfig.AdditionalFile.Path)} already exists. Please remove '{conflictingAdditionalFilePath}' and try again.");
```

Comment:
Throw `SonarLintException` and localize the error message.
Also, I'd put single quotes round the project name, and put a line break after the first sentence (more readable, and consistent with the new error messages I've just added).

FYI it looks like release builds of SLVS only log the exception message, not the full stack trace, so I don't think we need to do anything special to reduce the noise in the output window.